maintaining-and-tuning/monitoring.md

```diff
@@ -198,16 +224,160 @@ These instructions assume that GRR server and Prometheus are both up and running
-If you want to disable or remove an alert, go to the dashboard and the corresponding panel, and there you can remove the alerting rule.
-
-1. You can now use the dashboards. The dashboards can give a general overview over the main components of the GRR server and Fleetspeak servers, which can be utilized by the user to monitor different metrics of each component. Examples for such metrics can be found in the [examples above](#example-queries). Remember that the dashboards and alerts are flexible, and can be expanded or modified to adjust to your exact needs. Additional metrics can be used by exploring `http://<host>:<port>/metrics` for each component of GRR server (change the port according to the GRR server component you want) and Fleetspeak servers, and if you'd like to create your own [custom dashboards](https://grafana.com/docs/grafana/latest/getting-started/getting-started/#create-a-dashboard), [panels](https://grafana.com/docs/grafana/latest/panels/add-a-panel/#add-a-panel) and [alerts](https://grafana.com/docs/grafana/latest/alerting/create-alerts/#create-alerts), make sure to go over the corresponding documentation in Grafana.
+1. Grafana is now set up and can be visited at `http://localhost:3000`.
+The username and password should be "admin"; please change it.
+If the Grafana UI doesn't show up, either the
+[installation of Grafana](https://grafana.com/docs/grafana/latest/installation/#install-grafana)
+or the [server run](https://grafana.com/docs/grafana/latest/installation/debian/#2-start-the-server)
+failed; make sure to check the official documentation.
+
+1. [Set up Prometheus as a data source](https://grafana.com/docs/grafana/latest/features/datasources/prometheus/#prometheus-data-source)
+for Grafana, so that Grafana can display the metrics from Prometheus.
+To do that, follow the
+[these instructions](https://grafana.com/docs/grafana/latest/features/datasources/add-a-data-source/#add-a-data-source):
+Click on Configuration -> Data Sources in the Grafana server UI, then click
+on the "Add data source" button. From the list of "time series databases", choose
+Prometheus and in the Name field type `grr-server` (for Fleetspeak, type `fleetspeak`).
```

Comment:
Should both data sources be added for FS-based deployments?

Reply:
The [sample dashboards](https://github.com/google/grr/tree/master/monitoring/grafana/grr_grafanalib_dashboards) are designed in a way that Fleetspeak can be monitored independently from GRR. If that was the case, then we'd need two separate data sources for GRR and Fleetspeak separately.
In retrospect, that isn't what we aim for. I'll change the documentation here to refer to only one data source `grr-server` and change the [sample dashboard for Fleetspeak](https://github.com/google/grr/blob/fb203630e3e90a40166e9529be8d74dcfbfec76d/monitoring/grafana/grr_grafanalib_dashboards/fleetspeak_enabled_setup/dashboards_to_generate/fleetspeak.dashboard.py#L263) accordingly to reflect this.

Reply:
google/grr#896 references this.